src/components/DeleteAccountModal/DeleteAccountModal.tsx

```diff
@@ -0,0 +1,103 @@
+import { useTranslation } from 'react-i18next';
+import { type SchemaOf, object, string } from 'yup';
+import { useNavigate, useLocation } from 'react-router';
+import { useState } from 'react';
+import { useMutation } from 'react-query';
+
+import PasswordField from '../PasswordField/PasswordField';
+import Button from '../Button/Button';
+
+import styles from './DeleteAccountModal.module.scss';
+
+import type { DeleteAccountFormData } from '#types/account';
+import useForm from '#src/hooks/useForm';
+import { addQueryParam, removeQueryParam } from '#src/utils/location';
+import { deleteAccountData, logout } from '#src/stores/AccountController';
+import Alert from '#components/Alert/Alert';
+
+const DeleteAccountModal = () => {
+  const { t } = useTranslation('user');
+
+  const [enteredPassword, setEnteredPassword] = useState<string>('');
+
+  const deleteAccount = useMutation(deleteAccountData, {
+    onSuccess: async () => {
+      await logout();
+      navigate('/');
+    },
+    onError: () => {
+      setEnteredPassword('');
+    },
+  });
+
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const validationSchema: SchemaOf<DeleteAccountFormData> = object().shape({
+    password: string().required(t('login.field_required')),
+  });
+  const initialValues: DeleteAccountFormData = { password: '' };
+  const { handleSubmit, handleChange, values, errors } = useForm(
+    initialValues,
+    () => {
+      setEnteredPassword(values.password);
+      navigate(addQueryParam(location, 'u', 'delete-account-confirmation'));
+    },
+    validationSchema,
+  );
+
+  const handleCancel = () => {
+    navigate(removeQueryParam(location, 'u'));
+  };
+
+  if (deleteAccount.isError) {
+    return <Alert open isSuccess={false} onClose={handleCancel} message={t('account.delete_account.error')} />;
+  }
```

Comment:
Minor and opinionated, but perhaps sending the user back to the first step is more user-friendly instead of closing the modal? 

```suggestion
  const handleCancel = () => {
    navigate(removeQueryParam(location, 'u', { replace: true }));
  };
  
  const handleAlertClose = () => {
    // reset password (perhaps a state reset of the form is also needed)
    setEnteredPassword('');
    // go back one step
    navigate(addQueryParam(location, 'u', 'delete-account'), { replace: true });
  };

  if (deleteAccount.isError) {
    return <Alert open isSuccess={false} onClose={handleCancel} message={t('account.delete_account.error')} />;
  }
```